src/ring_theory/simple_module.lean

```diff
@@ -0,0 +1,108 @@
+/-
+Copyright (c) 2020 Aaron Anderson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors : Aaron Anderson
+-/
+
+import linear_algebra.basic
+
+/-!
+# Simple Modules
+
+## Main Definitions
+  * `is_simple_module` indicates that a module has no proper submodules
+  (the only submodules are `⊥` and `⊤`).
+  * A `division_ring` structure on the endomorphism ring of a simple module.
+
+## Main Results
+  * Schur's Lemma: `bijective_or_eq_zero_of_simple` shows that a linear map between simple modules
+  is either bijective or 0, leading to a `division_ring` structure on the endomorphism ring.
+
+## TODO
+  * Semisimple modules, Artin-Wedderburn Theory
+  * Unify with the work on Schur's Lemma in a category theory context
+
+-/
+
+variables (R : Type*) [comm_ring R] (M : Type*) [add_comm_group M] [module R M]
+
+/-- A module is simple when it has only two submodules, `⊥` and `⊤`. -/
+abbreviation is_simple_module := (is_simple_lattice (submodule R M))
+
+-- Making this an instance causes the linter to complain of "dangerous instances"
+theorem is_simple_module.nontrivial [is_simple_module R M] : nontrivial M :=
+⟨⟨0, begin
+  have h : (⊥ : submodule R M) ≠ ⊤ := bot_ne_top,
+  contrapose! h,
+  ext,
+  simp [submodule.mem_bot,submodule.mem_top, h x],
+end⟩⟩
+
+variables {R} {M}  {N : Type*} [add_comm_group N] [module R N]
+
+namespace linear_map
+
+theorem injective_or_eq_zero_of_simple [is_simple_module R M] (f : M →ₗ[R] N) :
```

Comment:
Personally I would drop the `of_simple` suffix on all these names. It's a typeclass implicit argument. Also, everybody who knows a bit of algebra will figure out that a linear map being `injective` or `eq_zero` will need some extra bit of input, namely simplicity.
In short: I don't think we loose much precision, and I think the brevity is worth it.